app/models/user.rb

```diff
@@ -18,6 +18,11 @@ def to_email_string
     end
   end
 
+  def sucard_number=(card_number)
+    return unless card_number.present?
+    self.library_id = card_number[/\d{5}(\d+)/, 1]
```

Comment:
My concern is that I don't know of those 5 digits (`80609`) are legitimately the only ones that will ever show up (another downside of this being completely undocumented information).